frontend/src/components/PackageGeneration/Editor.js

```diff
@@ -35,9 +35,31 @@ class editor extends React.Component {
     }
 
    componentDidMount() {
-    this.setState({code: localStorage.getItem("packageConfigYAML")})
-    this.setState({title: JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]["title"]})
-    this.setState({description: JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]["description"]})
+
+    if (localStorage.getItem("packageConfigYAML")) {
+      this.setState({code: localStorage.getItem("packageConfigYAML")})
+    } else {
+      this.setState({code: "Kindly generate or paste your configuration here"})
+    }
+
+    if (!localStorage.getItem("packageConfigJSON")["bundle"] === undefined ) {
+      if (localStorage.getItem("packageConfigJSON")["bundle"]["title"] === undefined ) {
+        this.setState({title: "No title Specified"})
+      }
+      else {
+        this.setState({title: JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]["title"]})
+     }
+   }
+
+    if (!localStorage.getItem("packageConfigJSON")["bundle"] === undefined ) {
+      if (localStorage.getItem("packageConfigJSON")["bundle"]["description"] === undefined ) {
+        this.setState({description: "No description specified"})
+      }
+      else {
+        this.setState({description: JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]["description"]})
+      }
+  }
```

Comment:
Should you also notes if there are issues if "bundle" isn't in the packageConfigJSON?

Reply:
Should I add another condition to check if the bundle is present ?

Reply:
Only if there's not a check earlier for if `JSON.parse(localStorage.getItem("packageConfigJSON"))["bundle"]` is missing.  If it's missing, then you can't do generation, but you should put something out to the console so you can trace where a problem happened.

Reply:
These are just for the description for the cards so we do not really care if it present or not. If the generation errors out the backend comes back with the 404.